qadence/ml_tools/train_grad.py

```diff
@@ -217,8 +240,19 @@ def loss_fn(model: torch.nn.Module, data: torch.Tensor) -> tuple[torch.Tensor, d
             except KeyboardInterrupt:
                 logger.info("Terminating training gracefully after the current iteration.")
                 break
+        try:
+            xs = next(dl_iter) if dataloader is not None else None  # type: ignore[arg-type]
+            xs_to_device = data_to_device(xs, device=device, dtype=data_dtype)
+            loss, metrics = loss_fn(model, xs_to_device)
+            if "val_loss" in metrics:
+                del metrics["val_loss"]
```

Comment:
@chMoussa could you please explain why the `val_loss` key-value pair is removed in this section and why the validation loss is being evaluated on the training dataset? I am not sure I understand the purpose of this.